src/Aspire.Hosting.Azure.Storage/AzureBlobStorageContainerResource.cs

```diff
@@ -0,0 +1,52 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Aspire.Hosting.ApplicationModel;
+using Aspire.Hosting.Azure;
+using Azure.Provisioning;
+
+namespace Aspire.Hosting;
+
+/// <summary>
+/// A resource that represents an Azure Blob Storage container.
+/// </summary>
+/// <param name="name">The name of the resource.</param>
+/// <param name="blobStorage">The <see cref="AzureBlobStorageResource"/> that the resource is stored in.</param>
+public class AzureBlobStorageContainerResource(string name, AzureBlobStorageResource blobStorage) : Resource(name),
+    IResourceWithConnectionString,
+    IResourceWithParent<AzureBlobStorageResource>,
+    IResourceWithAzureFunctionsConfig
+
+{
+    /// <summary>
+    /// Gets the connection string template for the manifest for the Azure Blob Storage container resource.
+    /// </summary>
+    public ReferenceExpression ConnectionStringExpression => Parent.GetConnectionString(Name);
+
+    /// <summary>
+    /// Gets the parent <see cref="AzureBlobStorageResource"/> of this <see cref="AzureBlobStorageContainerResource"/>.
+    /// </summary>
+    public AzureBlobStorageResource Parent => blobStorage ?? throw new ArgumentNullException(nameof(blobStorage));
+
+    internal void ApplyAzureFunctionsConfiguration(IDictionary<string, object> target, string connectionName)
+        => Parent.ApplyAzureFunctionsConfiguration(target, connectionName, Name);
+
+    /// <summary>
+    /// Converts the current instance to a provisioning entity.
+    /// </summary>
+    /// <returns>A <see cref="global::Azure.Provisioning.Storage.BlobContainer"/> instance.</returns>
+    internal global::Azure.Provisioning.Storage.BlobContainer ToProvisioningEntity()
```

Comment:
This is a new pattern, we don't do this anywhere else AFAIK.

Reply:
No, it's an existing pattern:
![image](https://github.com/user-attachments/assets/8bf0c942-7dfc-463c-8d7e-fdebec41608e)